### PR TITLE
Add more image augmentations: Verticalflip, colorjitter, randomaffine

### PR DESCRIPTION
**Pull request by Linuxdex**
*Issue #, if available:*

*Description of changes:*

Add verticalflip, colorjitter, randomaffine augmentations

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1719-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1719/1/index.html

**Comment by sxjscience** [comment]
Need to add more test-cases

**Comment by sxjscience** [comment]
Consider to add test-cases in https://github.com/awslabs/autogluon/tree/master/text/tests/unittests/automm . You may create a new file with name "test_data_process_image.py".

File: text/tests/unittests/automm/test_data_process_image.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "petfinder": PetFinderDataset,
+    "hateful_memes": HatefulMeMesDataset,
+    "ae": AEDataset,
+}
+
+@pytest.mark.parametrize(
+    "augmentations",
+    [
+        {
+            "model.timm_image.train_transform_types": ["resize_to_square","center_crop"],
```
Comment (sxjscience) [inline text/tests/unittests/automm/test_data_process_image.py:21]: Should we also test for randaug?

File: text/tests/unittests/automm/test_data_process_image.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            "model.timm_image.val_transform_types": ["resize_shorter_side","center_crop"]
+        },
+
+        {
+            "model.timm_image.train_transform_types": ["resize_shorter_side","center_crop","affine"],
+            "model.timm_image.val_transform_types": ["resize_shorter_side","center_crop"]
+        },
+
+        {
+            "model.timm_image.train_transform_types": ["resize_shorter_side","center_crop","colorjitter"],
```
Comment (sxjscience) [inline text/tests/unittests/automm/test_data_process_image.py:41]: Change to color_jitter

**Comment by szha** [comment]
Job PR-1719-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1719/6/index.html

File: text/src/autogluon/text/automm/data/process_image.py [hunk 3]
```diff
@@ -258,6 +258,12 @@ def construct_processor(
...
             elif trans_type == "horizontal_flip":
                 processor.append(transforms.RandomHorizontalFlip())
+            elif trans_type == "vertical_flip":
+                processor.append(transforms.RandomVerticalFlip())
+            elif trans_type == "color_jitter":
+                processor.append(transforms.ColorJitter(brightness=0.1, contrast=0.1, saturation=0.1))
+            elif trans_type == "affine":
+                processor.append(transforms.RandomAffine(15, translate=(0.1, 0.1), scale=(0.9, 1.1)))
             elif trans_type == "randaug":
                 processor.append(RandAugment(2, 9))
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/data/process_image.py:268]: The same for randaug. We can support `randaug` which means `RandAugment(2, 9)` and also support `randaug(2, 7)` or something similar.

**Comment by szha** [comment]
Job PR-1719-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1719/8/index.html

File: text/src/autogluon/text/automm/data/process_image.py [hunk 4]
```diff
@@ -250,18 +251,40 @@ def construct_processor(
         """
         processor = []
         for trans_type in transform_types:
-            if trans_type == "resize_to_square":
+            if '(' in trans_type:
+                trans_mode = trans_type[0:trans_type.find('(')]
+                arg = ast.literal_eval(trans_type[trans_type.find('('):])
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/data/process_image.py:256]: Use args

File: text/tests/unittests/automm/test_data_process_image.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        prefix="timm_image",
+        train_transform_types=augmentations["model.timm_image.train_transform_types"],
+        val_transform_types=augmentations["model.timm_image.val_transform_types"],
+        size=224,
+        norm_type="imagenet",
+    )
+
+    transimage = data_processors.train_processor(image)
+
+    assert len(data_processors.train_processor.transforms) == len(augmentations["model.timm_image.train_transform_types"])+2
```
Comment (sxjscience) [inline text/tests/unittests/automm/test_data_process_image.py:72]: Add a newline at the end of file.

File: text/tests/unittests/automm/test_data_process_image.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
```
Comment (sxjscience) [inline text/tests/unittests/automm/test_data_process_image.py:73]: You will need to remove all the spaces.

**Comment by szha** [comment]
Job PR-1719-9 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1719/9/index.html

**Review by gradientsky — APPROVED** [review]

**Review by bryanyzhu — APPROVED** [review]

**Comment by szha** [comment]
Job PR-1719-10 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1719/10/index.html

**Pull request merged**